geoindicators/src/main/groovy/org/orbisgis/orbisprocess/geoclimate/geoindicators/SpatialUnits.groovy

```diff
@@ -432,4 +444,84 @@ IProcess spatialJoin() {
             [outputTableName: outputTableName, idColumnTarget: idColumnTarget]
         }
     }
+}
+
+/**
+ * This process is used to generate a regular grid.
+ *
+ * @param geometry A geometry that defines either Point, Line or Polygon
+ * @param deltaX An integer that represents the spatial horizontal step of a cell in the mesh
+ * @param deltaY An integer that represents the spatial vertical step of a cell in the mesh
+ * @param outputTable A Table that contains the geometry of the regular grid
+ * @param datasource A connexion to a database (H2GIS, POSTGIS, ...) where are stored the input Table and in which
+ *        the resulting database will be stored
+ * @param outputTableName The name of the created table
+ * */
+IProcess regularGrid() {
+    return create {
+        title "Creating a regular grid in meters"
+        id "regularGrid"
+        inputs geometry: Geometry, deltaX: double, deltaY: double, tableName: String, datasource: JdbcDataSource
+        outputs outputTableName: String
+
+        run { geometry, deltaX, deltaY, tableName, datasource ->
+            if (datasource.hasTable(tableName)) {
+                info "Table $tableName already exists"
+                return null
+            }
+            if (datasource instanceof H2GIS) {
+                info "Creating a regular grid with H2GIS"
+                datasource """CREATE TABLE $tableName AS SELECT * FROM 
+                                     ST_MakeGrid('$geometry'::geometry, $deltaX, $deltaY);
+                           """
+            }
+            else if (datasource instanceof POSTGIS) {
+                info "Creating a regular grid with POSTGIS"
+                    PreparedStatement preparedStatement = null
+                    Connection outputConnection = datasource.getConnection()
+                    try {
+                        def createTable = "CREATE TABLE $tableName(THE_GEOM GEOMETRY(POLYGON), ID INT, ID_COL INT, ID_ROW INT);"
+                        def insertTable = "INSERT INTO $tableName VALUES (?, ?, ?, ?);"
+
+                        datasource.execute(createTable)
+                        preparedStatement = outputConnection.prepareStatement(insertTable)
+
+                        def result = ST_MakeGrid.createGrid(outputConnection, ValueGeometry.getFromGeometry(geometry), deltaX, deltaY)
+
+                        long batch_size = 0
+                        int batchSize = 1000
+
+                        while (result.next()) {
+                            def geom = result.getObject(1)
+                            def id = result.getInt(2)
+                            def id_col = result.getInt(3)
+                            def id_row = result.getInt(4)
+                            preparedStatement.setObject( 1, geom)
```

Comment:
What about 
 preparedStatement.setObject( 1, result.getObject(1))

Reply:
Yes it works with this syntax, so I could do the same for the three other attributes (id, id_col, id_row) to avoid 'def' assignement